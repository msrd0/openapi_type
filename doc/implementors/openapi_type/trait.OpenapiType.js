(function() {var implementors = {};
implementors["openapi_type"] = [];
if (window.register_implementors) {window.register_implementors(implementors);} else {window.pending_implementors = implementors;}})()