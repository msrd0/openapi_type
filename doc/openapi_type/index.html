<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="This crate gives static type information for primitives and commonly used types from the standard library and other commonly used libraries `chrono`, `indexmap`, `linked-hash-map`, `time` and `uuid` when the according feature is enabled. Also, it provides a derive macro for structs and enums to gain access to their static type information at runtime."><meta name="keywords" content="rust, rustlang, rust-lang, openapi_type"><title>openapi_type - Rust</title><link rel="stylesheet" type="text/css" href="../normalize.css"><link rel="stylesheet" type="text/css" href="../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../light.css"  id="themeStyle"><link rel="stylesheet" type="text/css" href="../dark.css" disabled ><link rel="stylesheet" type="text/css" href="../ayu.css" disabled ><script id="default-settings" ></script><script src="../storage.js"></script><script src="../crates.js"></script><script defer src="../main.js"></script>
    <noscript><link rel="stylesheet" href="../noscript.css"></noscript><link rel="alternate icon" type="image/png" href="../favicon-16x16.png"><link rel="alternate icon" type="image/png" href="../favicon-32x32.png"><link rel="icon" type="image/svg+xml" href="../favicon.svg"><style type="text/css">#crate-search{background-image:url("../down-arrow.svg");}</style></head><body class="rustdoc mod crate"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="sidebar"><div class="sidebar-menu" role="button">&#9776;</div><a href='../openapi_type/index.html'><div class='logo-container rust-logo'><img src='../rust-logo.png' alt='logo'></div></a><h2 class="location">Crate openapi_type</h2><div class="block version"><div class="narrow-helper"></div><p>Version 0.4.0</p></div><div class="sidebar-elems"><a id="all-types" href="all.html"><p>See all openapi_type's items</p></a><div class="block items"><ul><li><a href="#structs">Structs</a></li><li><a href="#enums">Enums</a></li><li><a href="#traits">Traits</a></li></ul></div><div id="sidebar-vars" data-name="openapi_type" data-ty="mod" data-relpath=""></div><script defer src="sidebar-items.js"></script></div></nav><div class="theme-picker"><button id="theme-picker" aria-label="Pick another theme!" aria-haspopup="menu" title="themes"><img width="18" height="18" alt="Pick another theme!" src="../brush.svg"></button><div id="theme-choices" role="menu"></div></div><nav class="sub"><form class="search-form"><div class="search-container"><div><select id="crate-search"><option value="All crates">All crates</option></select><input class="search-input" name="search" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"></div><button type="button" id="help-button" title="help">?</button><a id="settings-menu" href="../settings.html" title="settings"><img width="18" height="18" alt="Change settings" src="../wheel.svg"></a></div></form></nav><section id="main" class="content"><h1 class="fqn"><span class="in-band">Crate <a class="mod" href="#">openapi_type</a><button id="copy-path" onclick="copy_path(this)" title="Copy item path to clipboard"><img src="../clipboard.svg" width="19" height="18" alt="Copy item path"></button></span><span class="out-of-band"><span id="render-detail"><a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span><a class="srclink" href="../src/openapi_type/lib.rs.html#1-108" title="goto source code">[src]</a></span></h1><details class="rustdoc-toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>This crate gives static type information for primitives and commonly used types from the standard
library and other commonly used libraries <code>chrono</code>, <code>indexmap</code>, <code>linked-hash-map</code>, <code>time</code> and
<code>uuid</code> when the according feature is enabled. Also, it provides a derive macro for structs and
enums to gain access to their static type information at runtime.</p>
<p>The core of this crate is the <a href="trait.OpenapiType.html" title="OpenapiType"><code>OpenapiType</code></a> trait. It has one static function,
<a href="trait.OpenapiType.html#method.schema"><code>schema</code></a>, which returns an <a href="struct.OpenapiSchema.html" title="OpenapiSchema"><code>OpenapiSchema</code></a>. This assembles the static
type information in a way that is convenient to use for a generated OpenAPI specification, but
can also be utilized in other use cases as well.</p>
<h2 id="custom-types" class="section-header"><a href="#custom-types">Custom Types</a></h2>
<p>To gain access to the static type information of your custom types at runtime, the easiest way
is to use the derive macro:</p>

<div class="example-wrap"><pre class="rust rust-example-rendered"><code><span class="attribute">#[<span class="ident">derive</span>(<span class="ident">OpenapiType</span>)]</span>
<span class="kw">struct</span> <span class="ident">FooBar</span> {
	<span class="ident">foo</span>: <span class="ident">String</span>,
	<span class="ident">bar</span>: <span class="ident">u64</span>
}</code></pre></div>
<h2 id="openapi-specification" class="section-header"><a href="#openapi-specification">OpenAPI specification</a></h2>
<p>Using above type, running <code>FooBar::schema().into_schema()</code> yields</p>
<div class="example-wrap"><pre class="language-yaml"><code>type: object
title: FooBar
properties:
  foo:
    type: string
  bar:
    type: integer
    format: int64
    minimum: 0
required:
  - foo
  - bar</code></pre></div>
<p>Note, however, that this is not sufficient for more complex types. If one of your structs fields
is a type that has a name (that is, <code>Type::schema().name</code> is not <code>None</code>), above schema will contain
a reference to that schema. Therefore, always remember to put the
<a href="struct.OpenapiSchema.html#structfield.dependencies"><code>dependencies</code></a> into the specification alongside the type you are
interested in.</p>
</div></details><h2 id="reexports" class="small-section-header"><a href="#reexports">Re-exports</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left import-item"><code>pub use <a class="mod" href="../indexmap/index.html" title="mod indexmap">indexmap</a>;</code></div><div class="item-right docblock-short"></div></div><div class="item-row"><div class="item-left import-item"><code>pub use <a class="mod" href="../openapiv3/index.html" title="mod openapiv3">openapiv3</a>;</code></div><div class="item-right docblock-short"></div></div></div><h2 id="structs" class="small-section-header"><a href="#structs">Structs</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.OpenapiSchema.html" title="openapi_type::OpenapiSchema struct">OpenapiSchema</a></div><div class="item-right docblock-short"></div></div></div><h2 id="enums" class="small-section-header"><a href="#enums">Enums</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.OpenapiVisitor.html" title="openapi_type::OpenapiVisitor enum">OpenapiVisitor</a></div><div class="item-right docblock-short"></div></div></div><h2 id="traits" class="small-section-header"><a href="#traits">Traits</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="trait" href="trait.AlternativesVisitor.html" title="openapi_type::AlternativesVisitor trait">AlternativesVisitor</a></div><div class="item-right docblock-short"></div></div><div class="item-row"><div class="item-left module-item"><a class="trait" href="trait.ObjectVisitor.html" title="openapi_type::ObjectVisitor trait">ObjectVisitor</a></div><div class="item-right docblock-short"></div></div><div class="item-row"><div class="item-left module-item"><a class="trait" href="trait.OpenapiType.html" title="openapi_type::OpenapiType trait">OpenapiType</a></div><div class="item-right docblock-short"><p>This trait needs to be implemented by every type that is being used in the OpenAPI Spec. It gives
access to the <a href="struct.OpenapiSchema.html" title="OpenapiSchema">OpenapiSchema</a> of this type. It is provided for primitive types, String and the
like. For use on your own types, there is a derive macro:</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="trait" href="trait.Visitor.html" title="openapi_type::Visitor trait">Visitor</a></div><div class="item-right docblock-short"><p>This trait can be used to visit a type. Call <strong>one</strong> of the methods on this
trait <strong>exactly once</strong>.</p>
</div></div></div><h2 id="derives" class="small-section-header"><a href="#derives">Derive Macros</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="derive" href="derive.OpenapiType.html" title="openapi_type::OpenapiType derive">OpenapiType</a></div><div class="item-right docblock-short"><p>The derive macro for <a href="https://docs.rs/openapi_type/*/openapi_type/trait.OpenapiType.html"><code>OpenapiType</code></a>.</p>
</div></div></div></section><section id="search" class="content hidden"></section><div id="rustdoc-vars" data-root-path="../" data-current-crate="openapi_type" data-search-index-js="../search-index.js" data-search-js="../search.js"></div>
</body></html>