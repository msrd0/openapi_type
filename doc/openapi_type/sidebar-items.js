window.SIDEBAR_ITEMS = {"derive":["OpenapiType"],"enum":["OpenapiVisitor"],"struct":["OpenapiSchema"],"trait":["AlternativesVisitor","ObjectVisitor","OpenapiType","Visitor"]};