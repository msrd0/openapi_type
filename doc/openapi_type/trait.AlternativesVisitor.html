<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="API documentation for the Rust `AlternativesVisitor` trait in crate `openapi_type`."><meta name="keywords" content="rust, rustlang, rust-lang, AlternativesVisitor"><title>AlternativesVisitor in openapi_type - Rust</title><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceSerif4-Regular.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../FiraSans-Regular.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../FiraSans-Medium.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceCodePro-Regular.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceSerif4-Bold.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceCodePro-Semibold.ttf.woff2"><link rel="stylesheet" type="text/css" href="../normalize.css"><link rel="stylesheet" type="text/css" href="../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../ayu.css" disabled><link rel="stylesheet" type="text/css" href="../dark.css" disabled><link rel="stylesheet" type="text/css" href="../light.css" id="themeStyle"><script id="default-settings" ></script><script src="../storage.js"></script><script src="../crates.js"></script><script defer src="../main.js"></script>
    <noscript><link rel="stylesheet" href="../noscript.css"></noscript><link rel="alternate icon" type="image/png" href="../favicon-16x16.png"><link rel="alternate icon" type="image/png" href="../favicon-32x32.png"><link rel="icon" type="image/svg+xml" href="../favicon.svg"></head><body class="rustdoc trait"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle">&#9776;</button><a class="sidebar-logo" href="../openapi_type/index.html"><div class="logo-container"><img class="rust-logo" src="../rust-logo.svg" alt="logo"></div>
        </a><h2 class="location"></h2>
    </nav>
    <nav class="sidebar"><a class="sidebar-logo" href="../openapi_type/index.html"><div class="logo-container"><img class="rust-logo" src="../rust-logo.svg" alt="logo"></div>
        </a><h2 class="location"><a href="#">AlternativesVisitor</a></h2><div class="sidebar-elems"><section><div class="block"><h3 class="sidebar-title"><a href="#required-associated-types">Required Associated Types</a></h3><ul><li><a href="#associatedtype.Visitor">Visitor</a></li></ul></div><div class="block"><h3 class="sidebar-title"><a href="#required-methods">Required Methods</a></h3><ul><li><a href="#tymethod.visit_alternative">visit_alternative</a></li><li><a href="#tymethod.visit_description">visit_description</a></li><li><a href="#tymethod.visit_name">visit_name</a></li></ul></div><div class="block"><h3 class="sidebar-title"><a href="#implementors">Implementors</a></h3></div></section><h2 class="location"><a href="index.html">In openapi_type</a></h2><div id="sidebar-vars" data-name="AlternativesVisitor" data-ty="trait" data-relpath=""></div><script defer src="sidebar-items.js"></script></div></nav><main><div class="width-limiter"><div class="sub-container"><a class="sub-logo-container" href="../openapi_type/index.html"><img class="rust-logo" src="../rust-logo.svg" alt="logo"></a><nav class="sub"><div class="theme-picker hidden"><button id="theme-picker" aria-label="Pick another theme!" aria-haspopup="menu" title="themes"><img width="22" height="22" alt="Pick another theme!" src="../brush.svg"></button><div id="theme-choices" role="menu"></div></div><form class="search-form"><div class="search-container"><span></span><input class="search-input" name="search" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"><button type="button" id="help-button" title="help">?</button><a id="settings-menu" href="../settings.html" title="settings"><img width="22" height="22" alt="Change settings" src="../wheel.svg"></a></div></form></nav></div><section id="main-content" class="content"><div class="main-heading">
    <h1 class="fqn"><span class="in-band">Trait <a href="index.html">openapi_type</a>::<wbr><a class="trait" href="#">AlternativesVisitor</a><button id="copy-path" onclick="copy_path(this)" title="Copy item path to clipboard"><img src="../clipboard.svg" width="19" height="18" alt="Copy item path"></button></span></h1><span class="out-of-band"><a class="srclink" href="../src/openapi_type/visitor/mod.rs.html#75-83">source</a> · <a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span></div><div class="docblock item-decl"><pre class="rust trait"><code>pub trait AlternativesVisitor: Sealed {
    type <a href="#associatedtype.Visitor" class="associatedtype">Visitor</a>: <a class="trait" href="trait.Visitor.html" title="trait openapi_type::Visitor">Visitor</a>;

    fn <a href="#tymethod.visit_name" class="fnname">visit_name</a>(&amp;mut self, name: <a class="struct" href="https://doc.rust-lang.org/1.62.1/alloc/string/struct.String.html" title="struct alloc::string::String">String</a>);
<span class="item-spacer"></span>    fn <a href="#tymethod.visit_description" class="fnname">visit_description</a>(&amp;mut self, description: <a class="struct" href="https://doc.rust-lang.org/1.62.1/alloc/string/struct.String.html" title="struct alloc::string::String">String</a>);
<span class="item-spacer"></span>    fn <a href="#tymethod.visit_alternative" class="fnname">visit_alternative</a>(&amp;mut self) -&gt; &amp;mut Self::<a class="associatedtype" href="trait.AlternativesVisitor.html#associatedtype.Visitor" title="type openapi_type::AlternativesVisitor::Visitor">Visitor</a>;
}</code></pre></div><h2 id="required-associated-types" class="small-section-header">Required Associated Types<a href="#required-associated-types" class="anchor"></a></h2><div class="methods"><div id="associatedtype.Visitor" class="method has-srclink"><div class="rightside"><a class="srclink" href="../src/openapi_type/visitor/mod.rs.html#76">source</a></div><h4 class="code-header">type <a href="#associatedtype.Visitor" class="associatedtype">Visitor</a>: <a class="trait" href="trait.Visitor.html" title="trait openapi_type::Visitor">Visitor</a></h4></div></div><h2 id="required-methods" class="small-section-header">Required Methods<a href="#required-methods" class="anchor"></a></h2><div class="methods"><div id="tymethod.visit_name" class="method has-srclink"><div class="rightside"><a class="srclink" href="../src/openapi_type/visitor/mod.rs.html#78">source</a></div><h4 class="code-header">fn <a href="#tymethod.visit_name" class="fnname">visit_name</a>(&amp;mut self, name: <a class="struct" href="https://doc.rust-lang.org/1.62.1/alloc/string/struct.String.html" title="struct alloc::string::String">String</a>)</h4></div><div id="tymethod.visit_description" class="method has-srclink"><div class="rightside"><a class="srclink" href="../src/openapi_type/visitor/mod.rs.html#80">source</a></div><h4 class="code-header">fn <a href="#tymethod.visit_description" class="fnname">visit_description</a>(&amp;mut self, description: <a class="struct" href="https://doc.rust-lang.org/1.62.1/alloc/string/struct.String.html" title="struct alloc::string::String">String</a>)</h4></div><div id="tymethod.visit_alternative" class="method has-srclink"><div class="rightside"><a class="srclink" href="../src/openapi_type/visitor/mod.rs.html#82">source</a></div><h4 class="code-header">fn <a href="#tymethod.visit_alternative" class="fnname">visit_alternative</a>(&amp;mut self) -&gt; &amp;mut Self::<a class="associatedtype" href="trait.AlternativesVisitor.html#associatedtype.Visitor" title="type openapi_type::AlternativesVisitor::Visitor">Visitor</a></h4></div></div><h2 id="implementors" class="small-section-header">Implementors<a href="#implementors" class="anchor"></a></h2><div class="item-list" id="implementors-list"></div><script type="text/javascript" src="../implementors/openapi_type/visitor/trait.AlternativesVisitor.js" data-ignore-extern-crates="" async></script></section></div></main><div id="rustdoc-vars" data-root-path="../" data-current-crate="openapi_type" data-themes="ayu,dark,light" data-resource-suffix="" data-rustdoc-version="1.62.1 (e092d0b6b 2022-07-16)" ></div>
</body></html>