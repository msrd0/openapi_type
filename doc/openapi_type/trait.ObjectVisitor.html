<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="API documentation for the Rust `ObjectVisitor` trait in crate `openapi_type`."><title>ObjectVisitor in openapi_type - Rust</title><link rel="preload" as="font" type="font/woff2" crossorigin href="../static.files/SourceSerif4-Regular-46f98efaafac5295.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../static.files/FiraSans-Regular-018c141bf0843ffd.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../static.files/FiraSans-Medium-8f9a781e4970d388.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../static.files/SourceCodePro-Regular-562dcc5011b6de7d.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../static.files/SourceSerif4-Bold-a2c9cd1067f8b328.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../static.files/SourceCodePro-Semibold-d899c5a5c4aeb14a.ttf.woff2"><link rel="stylesheet" href="../static.files/normalize-76eba96aa4d2e634.css"><link rel="stylesheet" href="../static.files/rustdoc-c4dbdcde0fbd8430.css" id="mainThemeStyle"><link rel="stylesheet" id="themeStyle" href="../static.files/light-db279b6232be9c13.css"><link rel="stylesheet" disabled href="../static.files/dark-cf923f49f397b216.css"><link rel="stylesheet" disabled href="../static.files/ayu-be46fdc453a55015.css"><script id="default-settings" 
        data-theme="ayu"
        data-use_system_theme="false"></script><script src="../static.files/storage-3891ce972e3a2bf8.js"></script><script defer src="sidebar-items.js"></script><script defer src="../static.files/main-98a684e84ae5b08b.js"></script><noscript><link rel="stylesheet" href="../static.files/noscript-13285aec31fa243e.css"></noscript><link rel="alternate icon" type="image/png" href="../static.files/favicon-16x16-8b506e7a72182f1c.png"><link rel="alternate icon" type="image/png" href="../static.files/favicon-32x32-422f7d1d52889060.png"><link rel="icon" type="image/svg+xml" href="../static.files/favicon-2c020d218678b618.svg"></head><body class="rustdoc trait"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle">&#9776;</button><a class="logo-container" href="../openapi_type/index.html"><img class="rust-logo" src="../static.files/rust-logo-151179464ae7ed46.svg" alt="logo"></a><h2></h2></nav><nav class="sidebar"><a class="logo-container" href="../openapi_type/index.html"><img class="rust-logo" src="../static.files/rust-logo-151179464ae7ed46.svg" alt="logo"></a><h2 class="location"><a href="#">ObjectVisitor</a></h2><div class="sidebar-elems"><section><h3><a href="#required-associated-types">Required Associated Types</a></h3><ul class="block"><li><a href="#associatedtype.FieldVisitor">FieldVisitor</a></li><li><a href="#associatedtype.FlattenVisitor">FlattenVisitor</a></li><li><a href="#associatedtype.ValueVisitor">ValueVisitor</a></li></ul><h3><a href="#required-methods">Required Methods</a></h3><ul class="block"><li><a href="#tymethod.visit_additional">visit_additional</a></li><li><a href="#tymethod.visit_deny_additional">visit_deny_additional</a></li><li><a href="#tymethod.visit_description">visit_description</a></li><li><a href="#tymethod.visit_field">visit_field</a></li><li><a href="#tymethod.visit_flatten_field">visit_flatten_field</a></li><li><a href="#tymethod.visit_name">visit_name</a></li></ul><h3><a href="#implementors">Implementors</a></h3></section><h2><a href="index.html">In openapi_type</a></h2></div></nav><main><div class="width-limiter"><nav class="sub"><form class="search-form"><span></span><input class="search-input" name="search" aria-label="Run search in the documentation" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"><div id="help-button" title="help" tabindex="-1"><a href="../help.html">?</a></div><div id="settings-menu" tabindex="-1"><a href="../settings.html" title="settings"><img width="22" height="22" alt="Change settings" src="../static.files/wheel-7b819b6101059cd0.svg"></a></div></form></nav><section id="main-content" class="content"><div class="main-heading"><h1>Trait <a href="index.html">openapi_type</a>::<wbr><a class="trait" href="#">ObjectVisitor</a><button id="copy-path" title="Copy item path to clipboard"><img src="../static.files/clipboard-7571035ce49a181d.svg" width="19" height="18" alt="Copy item path"></button></h1><span class="out-of-band"><a class="srclink" href="../src/openapi_type/visitor/mod.rs.html#57-73">source</a> · <button id="toggle-all-docs" title="collapse all docs">[<span>&#x2212;</span>]</button></span></div><pre class="rust item-decl"><code>pub trait ObjectVisitor: Sealed {
    type <a href="#associatedtype.FieldVisitor" class="associatedtype">FieldVisitor</a>: <a class="trait" href="trait.Visitor.html" title="trait openapi_type::Visitor">Visitor</a>;
    type <a href="#associatedtype.FlattenVisitor" class="associatedtype">FlattenVisitor</a>: <a class="trait" href="trait.Visitor.html" title="trait openapi_type::Visitor">Visitor</a>;
    type <a href="#associatedtype.ValueVisitor" class="associatedtype">ValueVisitor</a>: <a class="trait" href="trait.Visitor.html" title="trait openapi_type::Visitor">Visitor</a>;

    // Required methods
    fn <a href="#tymethod.visit_name" class="fn">visit_name</a>(&amp;mut self, name: <a class="struct" href="https://doc.rust-lang.org/1.69.0/alloc/string/struct.String.html" title="struct alloc::string::String">String</a>);
<span class="item-spacer"></span>    fn <a href="#tymethod.visit_description" class="fn">visit_description</a>(&amp;mut self, description: <a class="struct" href="https://doc.rust-lang.org/1.69.0/alloc/string/struct.String.html" title="struct alloc::string::String">String</a>);
<span class="item-spacer"></span>    fn <a href="#tymethod.visit_field" class="fn">visit_field</a>(
        &amp;mut self,
        name: <a class="struct" href="https://doc.rust-lang.org/1.69.0/alloc/string/struct.String.html" title="struct alloc::string::String">String</a>,
        doc: <a class="enum" href="https://doc.rust-lang.org/1.69.0/core/option/enum.Option.html" title="enum core::option::Option">Option</a>&lt;<a class="struct" href="https://doc.rust-lang.org/1.69.0/alloc/string/struct.String.html" title="struct alloc::string::String">String</a>&gt;
    ) -&gt; &amp;mut Self::<a class="associatedtype" href="trait.ObjectVisitor.html#associatedtype.FieldVisitor" title="type openapi_type::ObjectVisitor::FieldVisitor">FieldVisitor</a>;
<span class="item-spacer"></span>    fn <a href="#tymethod.visit_flatten_field" class="fn">visit_flatten_field</a>(&amp;mut self) -&gt; &amp;mut Self::<a class="associatedtype" href="trait.ObjectVisitor.html#associatedtype.FlattenVisitor" title="type openapi_type::ObjectVisitor::FlattenVisitor">FlattenVisitor</a>;
<span class="item-spacer"></span>    fn <a href="#tymethod.visit_deny_additional" class="fn">visit_deny_additional</a>(&amp;mut self);
<span class="item-spacer"></span>    fn <a href="#tymethod.visit_additional" class="fn">visit_additional</a>(&amp;mut self) -&gt; &amp;mut Self::<a class="associatedtype" href="trait.ObjectVisitor.html#associatedtype.ValueVisitor" title="type openapi_type::ObjectVisitor::ValueVisitor">ValueVisitor</a>;
}</code></pre><h2 id="required-associated-types" class="small-section-header">Required Associated Types<a href="#required-associated-types" class="anchor">§</a></h2><div class="methods"><section id="associatedtype.FieldVisitor" class="method"><a class="srclink rightside" href="../src/openapi_type/visitor/mod.rs.html#58">source</a><h4 class="code-header">type <a href="#associatedtype.FieldVisitor" class="associatedtype">FieldVisitor</a>: <a class="trait" href="trait.Visitor.html" title="trait openapi_type::Visitor">Visitor</a></h4></section><section id="associatedtype.FlattenVisitor" class="method"><a class="srclink rightside" href="../src/openapi_type/visitor/mod.rs.html#59">source</a><h4 class="code-header">type <a href="#associatedtype.FlattenVisitor" class="associatedtype">FlattenVisitor</a>: <a class="trait" href="trait.Visitor.html" title="trait openapi_type::Visitor">Visitor</a></h4></section><section id="associatedtype.ValueVisitor" class="method"><a class="srclink rightside" href="../src/openapi_type/visitor/mod.rs.html#60">source</a><h4 class="code-header">type <a href="#associatedtype.ValueVisitor" class="associatedtype">ValueVisitor</a>: <a class="trait" href="trait.Visitor.html" title="trait openapi_type::Visitor">Visitor</a></h4></section></div><h2 id="required-methods" class="small-section-header">Required Methods<a href="#required-methods" class="anchor">§</a></h2><div class="methods"><section id="tymethod.visit_name" class="method"><a class="srclink rightside" href="../src/openapi_type/visitor/mod.rs.html#62">source</a><h4 class="code-header">fn <a href="#tymethod.visit_name" class="fn">visit_name</a>(&amp;mut self, name: <a class="struct" href="https://doc.rust-lang.org/1.69.0/alloc/string/struct.String.html" title="struct alloc::string::String">String</a>)</h4></section><section id="tymethod.visit_description" class="method"><a class="srclink rightside" href="../src/openapi_type/visitor/mod.rs.html#64">source</a><h4 class="code-header">fn <a href="#tymethod.visit_description" class="fn">visit_description</a>(&amp;mut self, description: <a class="struct" href="https://doc.rust-lang.org/1.69.0/alloc/string/struct.String.html" title="struct alloc::string::String">String</a>)</h4></section><section id="tymethod.visit_field" class="method"><a class="srclink rightside" href="../src/openapi_type/visitor/mod.rs.html#66">source</a><h4 class="code-header">fn <a href="#tymethod.visit_field" class="fn">visit_field</a>(
    &amp;mut self,
    name: <a class="struct" href="https://doc.rust-lang.org/1.69.0/alloc/string/struct.String.html" title="struct alloc::string::String">String</a>,
    doc: <a class="enum" href="https://doc.rust-lang.org/1.69.0/core/option/enum.Option.html" title="enum core::option::Option">Option</a>&lt;<a class="struct" href="https://doc.rust-lang.org/1.69.0/alloc/string/struct.String.html" title="struct alloc::string::String">String</a>&gt;
) -&gt; &amp;mut Self::<a class="associatedtype" href="trait.ObjectVisitor.html#associatedtype.FieldVisitor" title="type openapi_type::ObjectVisitor::FieldVisitor">FieldVisitor</a></h4></section><section id="tymethod.visit_flatten_field" class="method"><a class="srclink rightside" href="../src/openapi_type/visitor/mod.rs.html#68">source</a><h4 class="code-header">fn <a href="#tymethod.visit_flatten_field" class="fn">visit_flatten_field</a>(&amp;mut self) -&gt; &amp;mut Self::<a class="associatedtype" href="trait.ObjectVisitor.html#associatedtype.FlattenVisitor" title="type openapi_type::ObjectVisitor::FlattenVisitor">FlattenVisitor</a></h4></section><section id="tymethod.visit_deny_additional" class="method"><a class="srclink rightside" href="../src/openapi_type/visitor/mod.rs.html#70">source</a><h4 class="code-header">fn <a href="#tymethod.visit_deny_additional" class="fn">visit_deny_additional</a>(&amp;mut self)</h4></section><section id="tymethod.visit_additional" class="method"><a class="srclink rightside" href="../src/openapi_type/visitor/mod.rs.html#72">source</a><h4 class="code-header">fn <a href="#tymethod.visit_additional" class="fn">visit_additional</a>(&amp;mut self) -&gt; &amp;mut Self::<a class="associatedtype" href="trait.ObjectVisitor.html#associatedtype.ValueVisitor" title="type openapi_type::ObjectVisitor::ValueVisitor">ValueVisitor</a></h4></section></div><h2 id="implementors" class="small-section-header">Implementors<a href="#implementors" class="anchor">§</a></h2><div id="implementors-list"></div><script src="../implementors/openapi_type/visitor/trait.ObjectVisitor.js" async></script></section></div></main><div id="rustdoc-vars" data-root-path="../" data-static-root-path="../static.files/" data-current-crate="openapi_type" data-themes="" data-resource-suffix="" data-rustdoc-version="1.69.0 (84c898d65 2023-04-16)" data-search-js="search-8a59a8356673ec1f.js" data-settings-js="settings-f0c5c39777a9a2f6.js" data-settings-css="settings-0bcba95ff279c1db.css" ></div></body></html>