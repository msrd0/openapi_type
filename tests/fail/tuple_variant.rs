use openapi_type::OpenapiType;

#[derive(OpenapiType)]
enum Foo {
	Pair(i64, i64)
}

fn main() {}
