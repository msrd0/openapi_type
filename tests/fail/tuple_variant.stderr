error: #[derive(OpenapiType)] does not support tuple variants
 --> $DIR/tuple_variant.rs:5:6
  |
5 |     Pair(i64, i64)
  |         ^^^^^^^^^^
